src/frontend/src/app/audit-log/audit-log.component.ts

```diff
@@ -24,6 +28,65 @@ export class AuditLogComponent implements OnInit {
     this.getAuditLogs();
   }
 
+  /**
+   *  Go to the page on which user clicked in pagination
+   */
+  goToPage(n: number): void {
+    this.page = n;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Go to the next page
+   */
+  onNext(): void {
+    this.page++;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Go to the previous page
+   */
+  onFirst(): void {
+    this.page = 1;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Go to the next page
+   */
+  onLast(n: number): void {
+    const p = Math.ceil(this.recordCount / this.limit) || 0;
+    this.page = p;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Go to the previous page
+   */
+  onPrev(): void {
+    this.page--;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Set limit and offset (it is internally called by goToPage(), onNext(), onPrev() methods)
```

Comment:
is it duplicate method?

Reply:
Not duplicate method in the same file (It is in different files). That method was in `assets.component.ts` and it is in `audit-log.component.ts`